Mathlib/NumberTheory/FactorisationProperties.lean

```diff
@@ -0,0 +1,320 @@
+  and `n` is positive -/
+def Deficient (n : ℕ) : Prop := ∑ i in properDivisors n, i < n
+
+/-- `n : ℕ` is Pseudoperfect if and only if there exists a subset of the proper divisors of n such
+  that the sum of that subset is equal to `n` and `n` is positive -/
+def Pseudoperfect (n : ℕ) : Prop :=
+  0 < n ∧ ∃ s : Finset ℕ, s ∈ powerset (properDivisors n) ∧ ∑ i in s, i = n
+
+/-- `n : ℕ` is a weird number if and only if it is both abundant and semi-perfect -/
+def WeirdNumber (n : ℕ) : Prop := Abundant n ∧ ¬ Pseudoperfect n
+
+theorem not_pseudoperfect_iff_forall :
+    ¬ Pseudoperfect n ↔ n = 0 ∨ ∀ (s : Finset ℕ), s ∈ powerset (properDivisors n)
+      → ∑ i in s, i ≠ n := by
+    have hn : (¬ 0 < n) ↔ n = 0 := by simp only [not_lt, nonpos_iff_eq_zero]
+    dsimp [Pseudoperfect]
+    rw [not_and_or, not_exists, hn]
+    constructor
+    · rintro (h1 | h2)
+      · tauto
+      · right
+        intro s
+        have hs : ¬(s ∈ powerset (properDivisors n) ∧ ∑ i in s, i = n) := by exact h2 s
+        rw [not_and] at hs
+        exact hs
+    · rintro (h1 | h2)
+      · tauto
+      · right
+        simp_rw [not_and]
+        exact h2
+
+theorem one_deficient : Nat.Deficient 1 := by
+  dsimp [Nat.Deficient]
+  norm_num
+
+theorem two_deficient : Nat.Deficient 2 := by
+  dsimp [Nat.Deficient]
+  norm_num
+
+theorem three_deficient : Nat.Deficient 3 := by
+  dsimp [Nat.Deficient]
+  norm_num
+
+theorem six_perfect : Nat.Perfect 6 := by
+  dsimp [Nat.Perfect]
+  rw [show properDivisors 6 = {1, 2, 3} by rfl]
+  norm_num
+
+theorem twelve_abundant : Nat.Abundant 12 := by
+  dsimp [Nat.Abundant]
+  rw [show properDivisors 12 = {1,2,3,4,6} by rfl]
+  norm_num
+
+theorem seventy_weird : Nat.WeirdNumber 70 := by
+  dsimp [Nat.WeirdNumber, Nat.Abundant]
+  rw [not_pseudoperfect_iff_forall]
+  have div70 : properDivisors 70 = {1, 2, 5, 7, 10, 14, 35} := by rfl
+  constructor
+  · rw [div70]
+    repeat norm_num
+  · rw [div70]
+    right
+    intro s hs
+    fin_cases hs <;> decide
+
+theorem divisor_subseteq_mul (hn : 0 < n) (hm : 0 < m) :
+    divisors n ∪ divisors m ⊆ divisors (n * m) := by
+  intro a ha
+  simp only [mem_union, mem_divisors] at ha ⊢
+  rcases ha with hn1 | hm1
+  · exact ⟨dvd_trans hn1.1 ⟨m, rfl⟩, by positivity⟩
+  · exact ⟨dvd_trans hm1.1 ⟨n, mul_comm ..⟩, by positivity⟩
+
+theorem prop_divisors_subseteq_mul (hn : 0 < n) (hm : 0 < m) :
+    properDivisors n ∪ properDivisors m ⊆ properDivisors (n*m) := by
+      refine subset_iff.mpr ?_
+      rintro a ha
+      have ha' : a ∈ properDivisors n ∨ a ∈ properDivisors m := by
+        simp_all only [gt_iff_lt, le_mul_iff_one_le_right,
+          le_mul_iff_one_le_left, mem_union, mem_properDivisors]
+      rcases ha' with hn1 | hm1
+      · have han : a ∣ n := by
+          simp_all only [mem_properDivisors, and_self]
+        refine mem_properDivisors.mpr ?_
+        constructor
+        exact Dvd.dvd.mul_right han m
```

Comment:
The `constructor` call is creating two sub-goals. Please indent them appropriately with a focusing dot `·` (inserted as `\.`), e.g. `· exact Dvd.dvd.mul_right han m` and also for the `refine`.